notification/src/main/java/com/amazon/opendistroforelasticsearch/alerting/destination/client/DestinationEmailClient.java

```diff
@@ -0,0 +1,108 @@
+/*
+ *   Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.alerting.destination.client;
+
+import com.amazon.opendistroforelasticsearch.alerting.destination.message.BaseMessage;
+import com.amazon.opendistroforelasticsearch.alerting.destination.message.EmailMessage;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import javax.mail.Authenticator;
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import javax.mail.Message;
+import javax.mail.MessagingException;
+import javax.mail.PasswordAuthentication;
+import javax.mail.Session;
+import javax.mail.Transport;
+
+/**
+ * This class handles the connections to the given Destination.
+ */
+public class DestinationEmailClient {
+
+    private static final Logger logger = LogManager.getLogger(DestinationEmailClient.class);
+    
+    public String execute(BaseMessage message) throws Exception {
+        if (message instanceof EmailMessage) {
+            EmailMessage emailMessage = (EmailMessage) message;
+            Session session = null;
+
+            Properties prop = new Properties();
+            prop.put("mail.transport.protocol", "smtp");
+            prop.put("mail.smtp.host", emailMessage.getHost());
+            prop.put("mail.smtp.port", emailMessage.getPort());
+
+            if (emailMessage.getUsername() != null && !emailMessage.getUsername().equals("".toCharArray())) {
```

Comment:
For the second part of the statement, you can make it more readable by changing it to: `!emailMessage.getUsername().toString().isEmpty()`

Reply:
I hadn't written this piece and I had considered making a similar comment in the past but I believe this was done intentionally to not call `toString` on a `SecureString` object since the character array could be compared.

The `toString` method of `SecureString` has a comment that `"This should only be used with APIs that do not take CharSequence"`. Not sure if that applies here, if you have a strong preference either way, I can change this to your suggestion.

Reply:
Yea then that is fair